packages/support/lib/env.js

```diff
@@ -0,0 +1,156 @@
+  'node_modules',
+  '.cache',
+  'appium',
+  MANIFEST_BASENAME,
+);
+
+/**
+ * Finds an installation of `appium` in dirf `cwd` directory.
+ * @param {string} cwd - Directory where Node's module resolution algorithm can find something called `appium`
+ * @returns {boolean}
+ */
+export function isLocalAppiumInstalled (cwd) {
+  try {
+    resolveFrom(cwd, 'appium');
+    return true;
+  } catch {}
+  return false;
+}
+
+/**
+ * Finds `appium` in a `package.json` file, if `cwd` contains a `package.json` file.
+ * @param {NormalizedPackageJson} [pkg] - Directory to search for `package.json` file
+ * @returns {string|undefined}
+ */
+function getAppiumDependencyFromPackage (pkg) {
+  return (
+    pkg?.dependencies?.appium ??
+    pkg?.devDependencies?.appium ??
+    pkg?.bundleDependencies?.appium
+  );
+}
+
+/**
+ * Attempt to read a `package.json` in `dir`.  If it doesn't exist, resolves w/ `undefined`.
+ */
+export const readPackageInDir = _.memoize(
+  /**
+   * @param {string} cwd
+   * @todo better error handling
+   * @returns {Promise<NormalizedPackageJson|undefined>}
+   */
+  async function readPackageInDir (cwd) {
+    return await readPackage({cwd});
+  },
+);
+/**
+ * Finds `appium` if installed locally _or_ if a dep in a local `package.json` (and just not installed yet)
+ */
+const getLocalAppiumInfo = _.memoize(
+  /**
+   * @param {NormalizedPackageJson} [pkg]
+   * @param {string} [cwd]
+   * @returns {Promise<LocalAppiumInfo>}
+   */
+  async (pkg, cwd = process.cwd()) => {
+    const [isLocalInstall, dependencyVersion] = await B.all([
+      isLocalAppiumInstalled(cwd),
+      getAppiumDependencyFromPackage(pkg),
+    ]);
+    return {isLocalInstall, cwd, dependencyVersion};
+  },
+);
+
+/**
+ * Determines location of Appium's "home" dir
+ *
+ * - If `APPIUM_HOME` is set in the environment, use that
+ * - If we have an `extensions.yaml` in
+ *   {@link DEFAULT_APPIUM_HOME `DEFAULT_APPIUM_HOME`}, then use that.
+ * - If we have `appium` installed as a dependency in a local project _and_ the
+ *   dependency itself does not begin with a `file:` URL (which should not be
+ *   used by any project other than `appium-monorepo`), use the local dir
+ * - Otherwise, use {@link DEFAULT_APPIUM_HOME `DEFAULT_APPIUM_HOME`}
+ *
+ * The result of this when `cwd` is the `appium-monorepo` working copy should be
+ * `DEFAULT_APPIUM_HOME`.
+ *
+ * To be clear: if `appium` is a non-`file:` dependency, we don't care if it
+ * hasn't been installed yet; we use `cwd` as `APPIUM_HOME` regardless.
+ */
+export const resolveAppiumHome = _.memoize(
+  /**
+   * @param {string} [cwd] - Current working directory.  _Must_ be absolute, if specified.
+   */
+  async (cwd) => {
+    if (cwd && !path.isAbsolute(cwd)) {
+      throw new TypeError('`cwd` parameter must be an absolute path');
+    }
+
+    if (process.env.APPIUM_HOME) {
+      return process.env.APPIUM_HOME;
+    }
+
+    try {
+      cwd = cwd ?? process.cwd();
```

Comment:
not sure about javascript, but in Java I always try to [avoid](https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern) reassigning of arg values and create a new variable instead

Reply:
I hear what you're saying, but I'm not really convinced by many of the arguments in that blog post.  L120 is no different than assigning a default value in the function signature.  Which.. I think I can do here, so I'm going to do that.